modules/private.php

```diff
@@ -0,0 +1,341 @@
+	 * Returns true for super admins; if you don't want that, use is_really_private_blog_user.
+	 *
+	 * @param int   $blog Current WordPress blod id.
+	 * @param Mixed $user Current WordPress user or user_id.
+	 */
+	static function is_private_blog_user( $blog, $user ) {
+		global $wpdb;
+
+		if ( ! is_object( $user ) ) {
+			$user = new WP_User( $user );
+		}
+
+		if ( ! $user->ID ) {
+			return false;
+		}
+
+		if ( is_numeric( $blog ) ) {
+			$blog_id = intval( $blog );
+		} elseif ( is_object( $blog ) ) {
+			$blog_id = $blog->blog_id;
+		} elseif ( is_string( $blog ) ) {
+			$blog    = get_blog_info( $blog, '/', 1 );
+			$blog_id = $blog->blog_id;
+		} else {
+			$blog_id = $wpdb->blogid;
+		}
+
+		// check if the user has read permissions.
+		$the_user = wp_clone( $user );
+		$the_user->for_site( $blog_id );
+
+
+		/**
+		 * Filter the capabilites a user needs to have to see the site
+		 *
+		 * @module sitemaps
+		 * @since 6.9
+		 *
+		 * @param string $cap The lowest capability a user needs to have
+		 */
+		$capability = apply_filters( 'jetpack_private_capability', 'read' );
+		return $the_user->has_cap( $capability );
+	}
+
+	/**
+	 * Hides the blog's name on the login form for private blogs.
+	 */
+	static function privatize_blog_maybe_mask_blog_name() {
+		add_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ), 3, 2 );
+	}
+
+	/**
+	 * Replaces the the blog's "name" value with "Protected Blog"
+	 *
+	 * @see privatize_blog_maybe_mask_blog_name()
+	 * @param mixed $value The requested non-URL site information.
+	 * @param mixed $what  Type of information requested.
+	 */
+	static function privatize_blog_mask_blog_name( $value, $what ) {
+		if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+			$value = __( 'Protected Blog', 'jetpack' );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Remove the privatize_blog_mask_blog_name filter
+	 */
+	static function remove_privatize_blog_mask_blog_name_filter() {
+		remove_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ) );
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	static function privatize_blog_comments( $comment ) {
+		Jetpack_Private::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 **/
+	static function privatize_blog_priv_selector() {
+		?>
+		<style>
+			.jetpack-private__setting-disabled {
+				font-weight: bold;	
+				padding: 10px;
+			}
+			.option-site-visibility fieldset {
+				display: none;
+			}
+		</style>
+		</fieldset>
+		<div class="jetpack-private__setting-disabled highlight">
+			<?php printf( __( 'This setting is ignored because you <a href="%s">made your site private</a>', 'jetpack' ), admin_url( 'admin.php?page=jetpack' ) . '#/traffic?term=private' ); ?>
```

Comment:
I believe this should now point to `#/security?term=private`.

Reply:
Implemented in [a12e42a](https://github.com/Automattic/jetpack/pull/10333/commits/a12e42ad186140e847bb5dc19859518092400402)